taf/api/repository.py

```diff
@@ -116,6 +117,42 @@ def create_repository(
         write=True,
         no_commit_warning=True,
     )
+
+    # Create the .git/hooks directory and add the pre-push hook script if it does not exist
+    hooks_dir = Path(auth_repo.path) / ".git" / "hooks"
+    hooks_dir.mkdir(parents=True, exist_ok=True)
+
+    pre_push_script = hooks_dir / "pre-push"
+    if not pre_push_script.exists():
+        pre_push_script_content = """#!/bin/bash
+
+# Path to the TAF CLI executable
+TAF_CLI="taf"
+
+# Run the TAF validation command
+$TAF_CLI repo validate
+VALIDATION_STATUS=$?
+
+# Check the validation status
+if [ $VALIDATION_STATUS -ne 0 ]; then
+  echo "TAF validation failed. Push aborted."
+  exit 1
+fi
+
+# Allow the push if validation passes
+exit 0
+"""
+
+        with open(pre_push_script, "w") as file:
```

Comment:
We generally use `pathlib.write_text(content)` or `.write_bytes(content)` to write to files